src/main/java/ee/carlrobert/codegpt/settings/state/OpenAISettingsState.java

```diff
@@ -5,24 +5,39 @@
 import com.intellij.openapi.components.State;
 import com.intellij.openapi.components.Storage;
 import com.intellij.util.xmlb.XmlSerializerUtil;
+import com.intellij.util.xmlb.annotations.Transient;
 import ee.carlrobert.codegpt.credentials.OpenAICredentialsManager;
-import ee.carlrobert.codegpt.settings.service.ServiceSelectionForm;
+import ee.carlrobert.codegpt.settings.service.OpenAiServiceForm;
+import ee.carlrobert.codegpt.settings.state.util.RemoteOpenAiSettings;
 import ee.carlrobert.llm.client.openai.completion.OpenAIChatCompletionModel;
 import org.jetbrains.annotations.NotNull;
 
 @State(name = "CodeGPT_OpenAISettings_210", storages = @Storage("CodeGPT_OpenAISettings_210.xml"))
-public class OpenAISettingsState implements PersistentStateComponent<OpenAISettingsState> {
+public class OpenAISettingsState extends
+    RemoteOpenAiSettings<OpenAICredentialsManager> implements
+    PersistentStateComponent<OpenAISettingsState> {
 
   private static final String BASE_PATH = "/v1/chat/completions";
 
   private String organization = "";
-  private String baseHost = "https://api.openai.com";
-  private String path = BASE_PATH;
-  private String model = OpenAIChatCompletionModel.GPT_3_5.getCode();
   private boolean openAIQuotaExceeded;
 
+  public OpenAISettingsState() {
+    super("https://api.openai.com", BASE_PATH, OpenAIChatCompletionModel.GPT_3_5,
+        new OpenAICredentialsManager());
+  }
+
+  public OpenAISettingsState(String baseHost, String path, OpenAIChatCompletionModel model,
```

Comment:
I'm not sure if manually creating the instance is the right way to go. Since this is an application-level service, only one instance should be created, and which can be accessed via its static `getInstance()` method.

Reply:
JetBrains recommends using a [separate state class](https://plugins.jetbrains.com/docs/intellij/persisting-state-of-components.html#d264d665_40) for persistent components. Seems useful, especially when we're planning to change the entire structure anyway.

Reply:
True, I now separated the storage and state classes for every service, e.g:

https://github.com/carlrobertoh/CodeGPT/blob/c4e1f980aaa2f3330b768a7d7a53c95662d84ce5/src/main/java/ee/carlrobert/codegpt/settings/state/llama/LlamaSettingsState.java#L8-L14